source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -133,27 +134,51 @@ createHeaderValidatorFactory([[maybe_unused]] const envoy::extensions::filters::
                              [[maybe_unused]] Server::Configuration::FactoryContext& context) {
 
   Http::HeaderValidatorFactoryPtr header_validator_factory;
-  if (config.has_typed_header_validation_config()) {
 #ifdef ENVOY_ENABLE_UHV
-    auto* factory = Envoy::Config::Utility::getFactory<Http::HeaderValidatorFactoryConfig>(
-        config.typed_header_validation_config());
-    if (!factory) {
-      throw EnvoyException(fmt::format("Header validator extension not found: '{}'",
-                                       config.typed_header_validation_config().name()));
-    }
+  ::envoy::config::core::v3::TypedExtensionConfig legacy_header_validator_config;
+  if (!config.has_typed_header_validation_config()) {
```

Comment:
comment here as well?
something along the lines of "if using UHV with no configuration, ensure the defaults match Envoy's original behavior"

Reply:
Done